src/skeymgmt.c

```diff
@@ -43,19 +41,35 @@ static void *p11prov_aes_import(void *provctx, int selection,
 
     p = OSSL_PARAM_locate_const(params, OSSL_SKEY_PARAM_RAW_BYTES);
     if (p) {
+        unsigned char *key = NULL;
+        size_t keylen = 0;
         int ret =
             OSSL_PARAM_get_octet_string_ptr(p, (const void **)&key, &keylen);
+
         if (ret != RET_OSSL_OK) {
             P11PROV_raise(ctx, CKR_KEY_INDIGESTIBLE, "Invalid data");
             return NULL;
         }
-    } else {
-        /* Not a digestible secret key */
-        P11PROV_raise(ctx, CKR_KEY_INDIGESTIBLE, "Raw Bytes param unavailable");
-        return NULL;
+        return p11prov_obj_import_secret_key(ctx, CKK_AES, key, keylen);
+    }
+
+    p = OSSL_PARAM_locate_const(params, OSSL_OBJECT_PARAM_REFERENCE);
+    if (p) {
+        void *obj = NULL;
+        size_t referenze_sz = 0;
+        int ret = OSSL_PARAM_get_octet_string_ptr(p, (const void **)&obj,
+                                                  &referenze_sz);
+
+        if (ret != RET_OSSL_OK) {
+            P11PROV_raise(ctx, CKR_KEY_INDIGESTIBLE, "Invalid data");
+            return NULL;
+        }
+        return p11prov_common_load(obj, keylen, CKK_AES);
```

Comment:
passing keylen here which does not exist in this code block

Reply:
Fixed